tests/common/test_progress.py

```diff
@@ -127,11 +127,12 @@ def test_progress_inline_fasttime(lt_ctx, monkeypatch):
     # through a call to finalize_task() on the main thread
     # This tests the fallback mechanism in case comms fail
     states = reporter._history
-    start_progress = ProgressState(0., 16, 0, 0, 2)
+    progress_id = states[0].progress_id
```

Comment:
It would be nice to add a test here (or in the general `Context` tests) that specifies a reporter via the `Context` API and checks that this reporter is actually used.

(sorry that my review is coming in bit-by-bit today, some things going on concurrently...)

Reply:
The new API is now used in all the tests but one, and the injected reporter is used.